### PR TITLE
feat: Add student role with minimal access in Superset

### PR DESCRIPTION
**Pull request by nandodev-net**
### Description:
This pull request introduces a new role for students, designed to be similar to the instructor role but with read-only access. The aim is to ensure students can only access and view assets for which they have permissions.

### Changes:

- Defined a base student role with minimal permissions.
- Automated role assignment for students upon login.

### Testing:

- Logged in as a student at Superset dashboard

Screenshot from 2023-08-24 13-41-54

Screenshot from 2023-08-24 13-42-20

Please review the changes and provide feedback.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @nandodev-net!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

**Comment by Ian2012** [comment]
You are missing the part where the role should be assigned automatically at login.

https://github.com/openedx/tutor-contrib-aspects/blob/8a5c23633f015d0e65f9aafed358658e16843882/tutoraspects/templates/aspects/apps/superset/pythonpath/openedx_sso_security_manager.py#L177-L178

You can use something similar to this:
```python
if roles:
    return roles
if not bool("{{SUPERSET_BLOCK_STUDENT_ACCESS}}"):
   return ["student", f"student - {language}"]
raise Exception(f"Student {username} tried to access Superset")
```

**Comment by Ian2012** [comment]
Also, as you are replicating the permissions for the instructor role. Can you create a template that holds the shared permissions for both roles and inject it where it corresponds?

**Comment by Ian2012** [comment]
cc @nandodev-net

File: tutoraspects/templates/aspects/apps/superset/pythonpath/superset_config_docker.py [hunk 1]
```diff
@@ -63,12 +63,14 @@
...
     "gamma": ["Gamma"],      # Course staff
     "instructor": ["{{SUPERSET_ROLES_MAPPING.instructor}}"], # Course instructors
+    "student": ["{{SUPERSET_ROLES_MAPPING.student}}"], # Course students
     "operator": ["{{SUPERSET_ROLES_MAPPING.operator}}"], # Installation operators
     "public": ["Public"],    # AKA anonymous users
 }
 
 for language in LANGUAGES.keys():
     AUTH_ROLES_MAPPING[f"instructor-{language}"] = [f"{{SUPERSET_ROLES_MAPPING.instructor}} - {language}"]
+    AUTH_ROLES_MAPPING[f"student-{language}"] = [f"{{SUPERSET_ROLES_MAPPING.student}} - {language}"]
```
Comment (Ian2012) [inline tutoraspects/templates/aspects/apps/superset/pythonpath/superset_config_docker.py:73]: @nandodev-net can you include the operator here too? Seems like it's missing

**Review by Ian2012 — CHANGES_REQUESTED** [review]

File: config.yml
Comment (Ian2012) [inline config.yml:1]: Please remove this file

**Review by Ian2012 — APPROVED** [review]

File: tutoraspects/templates/aspects/apps/superset/pythonpath/openedx_sso_security_manager.py [hunk 2]
```diff
@@ -174,9 +174,13 @@ def _get_user_roles(self, username, language):
...
                 roles = self.extra_get_user_roles(username, decoded_access_token)
-                if bool("{{SUPERSET_BLOCK_STUDENT_ACCESS}}") and not roles:
+                if roles:
+                    return roles
+                
+                if {{SUPERSET_BLOCK_STUDENT_ACCESS}}:
                     raise Exception(f"Student {username} tried to access Superset")
-            return roles if roles else []
+                else:
+                    return ["student", f"student - {language}"]
```
Comment (bmtcril) [inline tutoraspects/templates/aspects/apps/superset/pythonpath/openedx_sso_security_manager.py:183]: Other language roles don't have spaces around the dash

File: tutoraspects/templates/aspects/apps/superset/pythonpath/openedx_sso_security_manager.py [hunk 3]
```diff
@@ -174,9 +174,13 @@ def _get_user_roles(self, username, language):
                 return ["instructor", f"instructor-{language}"]
             else:
                 roles = self.extra_get_user_roles(username, decoded_access_token)
-                if bool("{{SUPERSET_BLOCK_STUDENT_ACCESS}}") and not roles:
+                if roles:
+                    return roles
```
Comment (bmtcril) [inline tutoraspects/templates/aspects/apps/superset/pythonpath/openedx_sso_security_manager.py:178]: I'm somewhat concerned about this logic. Having a `student` role returned here would override `SUPERSET_BLOCK_STUDENT_ACCESS`. Maybe not a big deal, but can we add a check for that and throw an exception if it happens?

**Review by bmtcril — APPROVED** [review]
Thanks @nandodev-net , looks good to me!

**Comment by openedx-webhooks** [comment]
@nandodev-net 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/tutor-contrib-aspects&entry.867055334=https://github.com/openedx/tutor-contrib-aspects/pull/307&entry.1484655318=https://github.com/nandodev-net&entry.752974735=2023-08-17+23:24&entry.1917517419=2023-08-25+12:42&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**